sdk/digitaltwins/Azure.DigitalTwins.Core/src/Queries/QueryBuilders/ADTQuery.cs

```diff
@@ -0,0 +1,104 @@
+{
+    // NOTE -- accessibility internal for now to bypass needing static.
+    /// <summary>
+    /// Custom ADT query builder class that facilitates building queries against an ADT instance.
+    /// </summary>
+    internal class AdtQuery
+    {
+        /// <summary>
+        /// Used to prevent needing static.
+        /// </summary>
+        protected string someString { get; set; }
+
+        /// <summary>
+        /// The substance of the query is broken down into clauses, stored in a list
+        /// for future parsing purposes.
+        /// </summary>
+        protected List<BaseClause> Clauses { get; set; } = new List<BaseClause>();
+
+        /// <summary>
+        /// Adds a new clause to the Clauses component. Internally called within this class by
+        /// public methods that correspond to different clauses themselves (see Select, From, Where).
+        /// </summary>
+        /// <param name="clause"> The new clause object. </param>
+        protected void AddQueryComponent(BaseClause clause)
+        {
+            Clauses.Add(clause);
+        }
+
+        /// <summary>
+        ///  Removes the most recently added component to the Clauses list.
+        /// </summary>
+        /// <returns> The updated list of clauses. </returns>
+        protected List<BaseClause> RemoveQueryComponent()
+        {
+            return Clauses;
+        }
+
+        /// <summary>
+        /// Called to add a select clause (and its corresponding argument) to the query.
+        /// </summary>
+        /// <param name="args"> The arguments that define what we select (eg. *). </param>
+        /// <returns> TODO. </returns>
+        public AdtQuerySelect Select(params string[] args)
+        {
+            Console.WriteLine(someString);
+            Console.WriteLine(args);
+            return new AdtQuerySelect();
+        }
+
+        /// <summary>
+        /// Used when applying the TOP() aggregate from the ADT query language. Same functionality as select
+        /// but inserts TOP() into the query structure as well.
+        /// </summary>
+        /// <param name="count"> The argument for TOP(), ie the number of instances to return. </param>
+        /// <returns> The ADTQuery object itself. </returns>
+        public AdtQuerySelect SelectTop(int count)
+        {
+            Console.WriteLine(someString);
+            Console.WriteLine(count);
+            return new AdtQuerySelect();
+        }
+
+        /// <summary>
+        /// Used when applying the COUNT() aggregate from the ADT query language. Similar to SelectTop().
+        /// </summary>
+        /// <param name="args"> The arguments that we define what we select (eg. *). </param>
+        /// <returns> The ADTQuery object itself. </returns>
+        public AdtQuerySelect SelectCount(params string[] args)
+        {
+            Console.WriteLine(someString);
+            Console.WriteLine(args);
+            return new AdtQuerySelect();
+        }
+
+        /// <summary>
```

Comment:
this can be removed from here. Since we will tackle logical operators in the Where class.